crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -890,6 +890,70 @@ impl<'w> EntityWorldMut<'w> {
         self
     }
 
+    /// Removes a dynamic [`Component`] from the entity.
+    ///
+    /// You should prefer to use the typed API [`EntityWorldMut::remove`] where possible.
+    ///
+    /// # Safety
+    ///
+    /// - [`ComponentId`] must be from the same world as [`EntityWorldMut`]
+    /// - [`OwningPtr`] must be a valid reference to the type represented by [`ComponentId`]
+    pub unsafe fn remove_by_id(&mut self, component_id: ComponentId) -> &mut Self {
+        let archetypes = &mut self.world.archetypes;
+        let storages = &mut self.world.storages;
+        let components = &mut self.world.components;
+        let entities = &mut self.world.entities;
+        let removed_components = &mut self.world.removed_components;
+
+        let bundles = &mut self.world.bundles;
+        let bundle_info = bundles.init_info::<EmptyBundle>(components, storages);
```

Comment:
I don't think it works to just put `EmptyBundle` here? I think you need to do something similar to `insert_by_id` and get bundle_info via `self.world.bundles.init_component_info(self.world.components, component_id)`.

The way you're doing it now I suspect you would end up with the same archetype and return line 925.